tests/storage/test_roommember.py

```diff
@@ -177,6 +177,16 @@ def test__null_byte_in_display_name_properly_handled(self):
             tok=self.t_alice,
         )
 
+        res = self.get_success(
+            self.store.db_pool.simple_select_onecol(
+                "room_memberships", {"user_id": "@alice:test"}, "display_name"
+            )
+        )
+        # verify that the display name was alice before change in membership
+        self.assertEqual(res[0], "alice")
+        # verify that it is now None
+        self.assertEqual(res[1], None)
```

Comment:
Bah, slight problem here is that this is *probably* fine for tests, but it's relying on the database returning the rows in insertion order.

SQLite secretly maintains a 'rowid' (row ID) per row and stores rows in a B* tree keyed by those, so I can see it returning them in insertion order.
I assume Postgres does something similar — basically, returning them in order of insertion is 'easiest' for this query (it'd be more work to jumble them up) so I reckon this test will work!
(If the database was using an index to retrieve the rows, then it would probably return them in order of that index)

However, nothing guarantees that these databases will do that. I think a notable time that databases will violate this insertion order is if some rows get deleted and the database shuffles things around (e.g. Postgres' VACUUM; SQLite can recycles rowids afaik and SQLite will generate rowids somewhat randomly if you reach the limit).

I was curious about SQLite's VACUUM and in my case it kept the order when vacuuming, seemingly just rewriting rowids, oh well (I was hoping for some rows to be dramatically reordered! :P).
```sql
sqlite> create table tab1 (val TEXT);
sqlite> insert into tab1 values ("a"), ("b"), ("c");
sqlite> select rowid, val from tab1;
1|a
2|b
3|c
sqlite> delete from tab1 where val = "b";
sqlite> insert into tab1 values ("d"), ("e"), ("f");
sqlite> select rowid, val from tab1;
1|a
3|c
4|d
5|e
6|f
sqlite> vacuum;
sqlite> select rowid, val from tab1;
1|a
2|c
3|d
4|e
5|f
```

Anyway, excuse the tangent, but this is the kind of thing that has the tendency to bite you in the bum :(.
I think what I'd do here is:
* select from `room_memberships` before changing Alice's name
  * check your preconditions *and make a note of the event_id*
* update the name
* select from `room_memberships` again, **filtering out the row with the event_id you had before** and check your postcondition

Check the expected number of rows at each step to make sure that if we get more rows for some reason, they don't cause confusion in the future when the test breaks and some poor soul has to figure out why :p.

I'd probably use a list comprehension to do the filtering in Python (mostly since doing it in SQL here is a pain and it's not as though it's performance sensitive since it's a test), something like that:
```python
rows = [
	row for row in rows if row["event_id"] != first_event_id
]
```